pmd-apex/src/main/resources/category/apex/bestpractices.xml

```diff
@@ -168,4 +168,47 @@ trigger Accounts on Account (before insert, before update, before delete, after
 ]]>
         </example>
     </rule>
+
+    <rule name="DebugsShouldUseLoggingLevel"
+          since="6.17.0"
+          language="apex"
+          message="Calls to System.debug should specify a logging level."
+          class="net.sourceforge.pmd.lang.apex.rule.ApexXPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_bestpractices.html#debugsshoulduselogginglevel">
+       <description>
+The first parameter of System.debug, when using the signature with two parameters, is a LoggingLevel enum.
+
+Having the Logging Level specified provides a cleaner log, and improves readability of it.
+       </description>
+       <priority>5</priority>
+       <properties>
+          <property name="strictMode" type="Boolean" value="false" description="If true, mark statements that use the DEBUG enum of LoggingLevel."/>
+          <property name="version" value="2.0"/>
+          <property name="xpath">
+             <value>
+<![CDATA[
+//MethodCallExpression
+  [count(ReferenceExpression[lower-case(@Image)='system']) = 1]
+  [count(VariableExpression) = 0 or (
+    count(VariableExpression) = 1 and VariableExpression[lower-case(@Image)='debug' and $strictMode]
+  )]
+]]>
+             </value>
+          </property>
+       </properties>
+       <example>
+<![CDATA[
+@isTest
+public class Foo {
+    @isTest
+    static void bar() {
+        System.debug('Hey this code executed.'); // not good
+        System.debug(LoggingLevel.WARN, 'Hey, something might be wrong.'); // good
+        System.debug(LoggingLevel.DEBUG, 'Hey, something happened.'); // strictly not good
```

Comment:
I don't understand this example: It's a System.debug call, that uses a log level. This is good, isn't it?

Reply:
Ah, I get it know. If the property "strictMode" is enabled, then we also flag this debug loggings... ok. I'd add the word "strictMode" to the comment, since "strictly not good" was not enough for me...